arch/arm/soc/st_stm32/common/Makefile

```diff
@@ -0,0 +1,4 @@
+obj-$(CONFIG_STM32_ARM_MPU_ENABLE) += arm_mpu_regions.o
+
+zephyr: $(KERNEL_HEX_NAME)
```

Comment:
Why is the hex image support moving here? If you want it common, do it as a separate patch/PR.